### PR TITLE
[NDMII-3023] enrich snmp metrics with device user tags

### PR DESCRIPTION
**Pull request by jedupau**
### What does this PR do?

Add the internal resource tag to enrich the snmp device metrics with the user tags from REDAPL
I checked that if there is the host tag and this tag, this won't duplicate the tags

### CODE REVIEW

File: releasenotes/notes/snmp-user-tags-540914b084cbdb92.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Rework the ability to tag snmp device metrics with user-defined tags.
```
Comment (buraizu) [inline releasenotes/notes/snmp-user-tags-540914b084cbdb92.yaml:11]:
```suggestion
    Rework the ability to tag SNMP device metrics with user-defined tags.
```

**Review by buraizu — APPROVED** [review]
Thanks, approving with a minor update requested

File: releasenotes/notes/snmp-user-tags-540914b084cbdb92.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Rework the ability to tag SNMP device metrics with user-defined tags.
```
Comment (TCheruy) [inline releasenotes/notes/snmp-user-tags-540914b084cbdb92.yaml:11]: I'm not sure we need a release note since this is purely internal

**Review by TCheruy — APPROVED** [review]

**Pull request merged**